emsdk.py

```diff
@@ -1243,10 +1243,17 @@ def find_latest_installed_tool(name):
 def emscripten_npm_install(tool, directory):
   node_tool = find_latest_installed_tool('node')
   if not node_tool:
-    print('Failed to run "npm ci" in installed Emscripten root directory ' + tool.installation_path() + '! Please install node.js first!')
-    return False
+    try:
+      subprocess.check_output(['npm' + ('.cmd' if WINDOWS else ''), '--version'], stderr=subprocess.STDOUT, universal_newlines=True)
+      node_path = os.path.dirname(shutil.which('npm' + ('.cmd' if WINDOWS else '')))
+    except subprocess.CalledProcessError:
```

Comment:
We have a local `which` function in this file which you can use rather than shtuil.which (since we want to support python2). 

You don't need the try/catch then either.   You an just do 'node_path = which ...` and then `if not node_path`

Reply:
Ok